keps/sig-api-machinery/20190802-dynamic-coordinated-storage-version.md

```diff
@@ -172,6 +172,24 @@ During bootstrap, for each resource, the API server
   to avoid conflicting with other API servers.
 * installs the resource handler.
 
+The above mentioned process requires an API server to update the storageVersion
+before accepting API requests. If we don't enforce this order, data encoded in
+an unexpected version can sneak into etcd. For example, an API server persists a
+write request encoded in an obsoleted version, then it crashes before it can
+update the storageVersion. The storage migrator has no way to detect this write.
+
+For the cmd/kube-apiserver binary, we plan to enforce this order by adding a new
+filter to the [handler chain][]. Before kube-aggregator, kube-apiserver, and
+apiextension-apiserver have registered the storage version of the built-in
+resources they host, this filter rejects any API requests that are not destined
+to the storageVersion API or not sent by the loopbackClient.
```

Comment:
Is the loopback client still a risk?  it's writes are no less real and the resources it writes are likely to be more important.

Reply:
I agree that the loopbackClient is equally real if not more. 

I was trying to say that only the writes to the storageVersion API issued by the loopbackClient should be admitted. No other clients need to write to the storageVersion API when the cmd/kube-apiserver binary is still bootstrapping.